### PR TITLE
feat: update interval in spec

### LINKED ISSUE #3325: Document the new interval package
The new interval package as defined in #3323 will need to be documented in the SPEC. 

We've also identified some edge behaviors that we can write end to end tests for in the flux repo and then document. Once we've done this we can work on getting these tests promoted to official documentation. 

DOD: 
- flux SPEC is updated with interval package behavior
- end to end tests written with comprehensive comments underlining each particular case
- consult with docs team to get test comments promoted to official docs

### PR DESCRIPTION
**Pull request by fchikwekwe**
Closes #3325 

### Done checklist
- [x] docs/SPEC.md updated
- [ ] Test cases written

### CODE REVIEW

File: docs/SPEC.md [hunk 1]
```diff
@@ -1235,8 +1235,8 @@ The generator is then used to produce the set of intervals.
 The set of intervals will include all intervals that intersect with the initial range of time.
 The `intervals` function is designed to be used with the `intervals` parameter of the `window` function.
 
-By default the end boundary of an interval will align with the Unix epoch (zero time) modified by the offset of the `location` option.
```
Comment (fchikwekwe) [inline docs/SPEC.md:1238]: I don't think we've ever implemented a location parameter.

File: docs/SPEC.md [hunk 2]
```diff
@@ -1251,11 +1251,9 @@ Intervals has the following parameters:
 | every  | duration                     | Every is the duration between starts of each of the intervals. Defaults to the value of the `period` duration.                                                                                                          |
 | period | duration                     | Period is the length of each interval. It can be negative, indicating the start and stop boundaries are reversed. Defaults to the value of the `every` duration.                                                        |
 | offset | duration                     | Offset is the duration by which to shift the window boundaries. It can be negative, indicating that the offset goes backwards in time. Defaults to 0, which will align window end boundaries with the `every` duration. |
-| filter | (interval: interval) => bool | Filter accepts an interval record and returns a boolean value. Defaults to include all intervals.                                                                                                                       |
```
Comment (fchikwekwe) [inline docs/SPEC.md:1254]: I also dont think that even previous versions of the intervals function used a `filter` param.

File: docs/SPEC.md [hunk 3]
```diff
@@ -1266,28 +1264,14 @@ The intervals function has the following signature:
 Examples:
 
     intervals(every:1h)                        // 1 hour intervals
-    intervals(every:1h, period:2h)             // 2 hour long intervals every 1 hour
+    intervals(every:1h, period:2h)             // overlapping; 2 hour long intervals every 1 hour
     intervals(every:1h, period:2h, offset:30m) // 2 hour long intervals every 1 hour starting at 30m past the hour
     intervals(every:1w, offset:1d)             // 1 week intervals starting on Monday (by default weeks start on Sunday)
     intervals(every:1d, period:-1h)            // the hour from 11PM - 12AM every night
+    intervals(every:1d, period:1h)             // underlapping; the hour from 12AM - 1AM every night
```
Comment (fchikwekwe) [inline docs/SPEC.md:1271]: highlighting some known edge cases.

File: docs/SPEC.md [hunk 4]
```diff
@@ -1266,28 +1264,14 @@ The intervals function has the following signature:
...
-        every:1d,
-        filter: (interval) => !(weekday(time: interval.start) in [Sunday, Saturday]),
-    )
-    // Work hours from 9AM - 5PM on work days.
-    intervals(
-        every:1d,
-        period:8h,
-        offset:9h,
-        filter:(interval) => !(weekday(time: interval.start) in [Sunday, Saturday]),
-    )
```
Comment (fchikwekwe) [inline docs/SPEC.md:1290]: without filter, these examples are irrelevant

File: docs/SPEC.md (hunk 1 above)
Comment (nathanielc) [reply docs/SPEC.md:1238]: We will implement it sometime. Its a TODO for now.

File: docs/SPEC.md [hunk 5]
```diff
@@ -1333,35 +1317,12 @@ Examples using known start and stop dates:
...
-    // 1 day intervals including only Sundays and Saturdays
-    weekdends = intervals(every:1d, filter: (interval) => weekday(time:interval.start) in [Sunday, Saturday])
-    // 1 week intervals
-    weeks = intervals(every:1w)
-    // 1 month interval
-    months = intervals(every:1mo)
-    // 3 month intervals
-    quarters = intervals(every:3mo)
-    // 1 year intervals
-    years = intervals(every:1y)
```
Comment (nathanielc) [inline docs/SPEC.md:1363]: Hmm should we add these to an `intervals` package?

File: stdlib/interval/interval_test.flux [hunk 6]
```diff
@@ -38,7 +38,7 @@ package interval_test
 import "experimental"
 import "interval"
 
-// these test cases cannot pass CI, so they're skipped for now. Once we're able to test a non table stream test case,
+// TODO(faith): these test cases cannot pass CI, so they're skipped for now. Once we're able to test a non table stream test case,
```
Comment (nathanielc) [inline stdlib/interval/interval_test.flux:41]: We actually now have the ability to do tests on non stream cases. Ask @rockstar or @jsternberg for direction.  Lets not block this PR on that work. Can you create a new issue to unskip these tests using the new Flux acceptance tests?

**Review by jsternberg — APPROVED** [review]

File: stdlib/interval/interval_test.flux (hunk 6 above)
Comment (fchikwekwe) [reply stdlib/interval/interval_test.flux:41]: oh yay! added this issue to the agenda https://github.com/influxdata/flux/issues/3526

**Comment by fchikwekwe** [comment]
@nathanielc https://github.com/influxdata/flux/pull/3525#discussion_r585742446 should I leave this in the SPEC as unimplemented then? Do we first need to have a discussion about how we would implement location/time zone info so that I can reference an issue here?

File: docs/SPEC.md (hunk 5 above)
Comment (fchikwekwe) [reply docs/SPEC.md:1363]: If we decide to keep the intervals function, then yeah I think that would be convenient. I can create an issue to add these functions in and document them.

**Comment by fchikwekwe** [comment]
After further discussion, we've decided that the interval package with the `intervals` function isn't that useful right now. We will revisit including it in the future if needed. This PR will now remove the package and the associated SPEC docs.

**Review by jsternberg — APPROVED** [review]

**Pull request merged**